sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java

```diff
@@ -89,4 +89,27 @@ CREATE TABLE series (
                 addSub);
         this.addRustTestCase("latenessTest", compiler, getCircuit(compiler), data);
     }
+
+    @Test
+    public void testJoin() {
+        String ddl = """
+            CREATE TABLE series (
+                    metadata VARCHAR,
+                    event_time TIMESTAMP NOT NULL LATENESS INTERVAL '1:00' HOURS TO MINUTES
+            );
+            
+            CREATE TABLE shift(
+                    person VARCHAR,
+                    on_call DATE
+            );
```

Comment:
This only tests whether the circuit compiles, but not whether the circuit and the join operator's state is bounded right?

Reply:
yes, there is no easy API to test that, and you'd have to feed lots of data anyway.
That should use the profiling API. 
Hopefully we'll write more end-to-end tests for that.